src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php

```diff
@@ -176,19 +179,28 @@ protected function getContainerBuilder()
             return $this->containerBuilder;
         }
 
-        if (!$this->getApplication()->getKernel()->isDebug()) {
+        $kernel = $this->getApplication()->getKernel();
+
+        if (!$kernel->isDebug()) {
             throw new \LogicException('Debug information about the container is only available in debug mode.');
         }
 
-        if (!is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
-            throw new \LogicException('Debug information about the container could not be found. Please clear the cache and try again.');
+        if (is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
+            (new XmlFileLoader($container = new ContainerBuilder(), new FileLocator()))->load($cachedFile);
+        } else {
+            $buildContainer = \Closure::bind(function () { return $this->buildContainer(); }, $kernel, get_class($kernel));
+            $container = $buildContainer();
+            $container->getCompilerPassConfig()->setRemovingPasses(array());
+            $container->compile();
+            $filesystem = new Filesystem();
+            try {
+                $filesystem->dumpFile($cachedFile, (new XmlDumper($container))->dump());
```

Comment:
parameter removal and compiler pass deprecation reverted, it now leverages the debug mode properly which isn't the case currently. It would be great if you could have a look